lib/resque/worker.rb

```diff
@@ -379,61 +403,30 @@ def unregister_worker
         job.fail(DirtyExit.new)
       end
 
-      redis.srem(:workers, self)
-      redis.del("worker:#{self}")
-      redis.del("worker:#{self}:started")
-
-      Stat.clear("processed:#{self}")
-      Stat.clear("failed:#{self}")
+      redis.pipelined do
```

Comment:
Similar-but-different to master's implementation [here](https://github.com/al2o3cr/resque/pull/2/files#diff-102541f6f2ba476fceb8e45917e63420R244)

TODO: what's the `HEARTBEAT_KEY` stuff in master doing? Would it be useful to add here?